esphome/components/touchscreen/touchscreen.h

```diff
@@ -47,10 +57,38 @@ class Touchscreen {
   /// Call this function to send touch points to the `on_touch` listener and the binary_sensors.
   void send_touch_(TouchPoint tp);
 
-  uint16_t display_width_;
-  uint16_t display_height_;
-  display::DisplayBuffer *display_;
-  TouchRotation rotation_;
+  uint16_t get_width_() {
+#ifdef USE_DISPLAY
+    return display_->get_width_internal();
+#else
+    return display_width_;
+#endif
+  }
+  uint16_t get_height_() {
+#ifdef USE_DISPLAY
+    return display_->get_height_internal();
+#else
+    return display_height_;
+#endif
+  }
+
+  uint16_t get_rotation_() {
```

Comment:
Why not return the enum type?
 ```suggestion
  TouchRotation get_rotation_() {
```

Reply:
thanks, you are right, i missed this. I was to busy try it to work without the need of a display